app/celery/tasks.py

```diff
@@ -740,11 +582,26 @@ def handle_batch_error_and_forward(
             # in the form of a dict
             if isinstance(template, tuple):
                 template = template[0]
-
-            save_fn.apply_async(
-                (service.id, notification_id, signed, None),
-                queue=choose_database_queue(template, service),
+            retry_msg = "{task} notification for job {job} row number {row} and notification id {notif} and max_retries are {max_retry}".format(
+                task=task.__name__,
+                job=notification.get("job", None),
+                row=notification.get("row_number", None),
+                notif=notification_id,
+                max_retry=task.max_retries,
             )
+            current_app.logger.warning("Retry " + retry_msg)
+            try:
+                if task.max_retries == 5 and len(signed_and_verified) != 1:
```

Comment:
Can you please add a comment on the reason why we want to catch the max retries at 5?

Did you test that `task.max_retries` decrease on every retry iteration?

Reply:
sorry forgot about the comment, will add

for the second - max_retries doesn't reduce on each retry iteration, it is stored elsewhere and isn't exposed to the task. I tested this manually and saw the max_retry stay the same but the number of times the retry was called was 5.

Reply:
but ah this is a good point cause now I can change the if statement. ok let me think about it

Reply:
added a comment and reran the tests.